contractions/contractions_FP.c

```diff
@@ -267,6 +267,39 @@ void density_density_1234_s0s0( bispinor ** propfields, int type_1234, _Complex
 
 }
 
+void average_scalar_in_space( _Complex double **results){
+
+   _Complex double *spacetrace=(_Complex double *)malloc(sizeof(_Complex double) *4*T_global);
+   *results=(_Complex double *)malloc(sizeof(_Complex double)*4*T_global);
+   if (*results == NULL){
+     printf("Not enough memory for the results\n");
+     exit(1);
+   }
+   for (int j = 0; j<4 ; ++j ){
+      for (int x0=0; x0<T_global; ++x0){
+         spacetrace[j*T_global+x0]=0.0;
+      }
+   }
+   for (int i=0; i<VOLUME;++i){
+     spacetrace[g_coord[i][TUP]]+= g_smeared_scalar_field[0][i];
```

Comment:
`TUP` ?

Why is `g_smeared_scalar_field` averaged (again) here?

Reply:
I asked for a contraction that has the 0 component of the scalar field average over the space. It was already computed somewhere else but it is not really time-consuming to recompute it, I think.

Reply:
Sure, that's clear, the spatially-averaged scalar fields are now written out as part of the contractions. What I find strange is that it's `g_smeared_scalar_field` which is spatial-volume averaged again here. I would have expected `g_scalar_field` here (where we store the hc_nn_lex field).

Reply:
Maybe I'm misunderstanding something...

Reply:
yes, sorry, this is a bug, I will correct it

Reply:
Could you also use `0` here instead of `TUP` ? I find this a bit confusing here because `TUP` simply happens to be `0` since it's the first "forward hop" direction.